### PR TITLE
adding relro to i and putting stuff in semiorder

### PR DESCRIPTION
**Pull request by oddcoder**
for sure not the smartest way to do it

### CODE REVIEW

**Comment by alvarofe** [comment]
why to show relro info within `i`, relro is specific to ELF iirc. I would let it as it is.

**Comment by Maijin** [comment]
@alvarofe just like we show signature or overlay in PE having relro definitely interesting but it must be elf specific yes.

**Comment by radare** [comment]
Agree, unless thats generic... what about making some fileformat specific fields? Or what about looking for a generic thing of relro for non-elf that applies in a similar way?

**Comment by oddcoder** [comment]
it will only be shown if the file is elf so there should be no issue with that ?

**Comment by Maijin** [comment]
There shouldn't be issue...

**Comment by Maijin** [comment]
image
We already have PE specifics

**Comment by radare** [comment]
Signed is not specific also applies on macho. The same goes for checksums which work on dex and wtf is overlay. 

All this info applies to other file formats. How does relro ?

**Comment by Maijin** [comment]
Same could be applied for tons of case for example Gameboy/NES/Megadrive/Menuet doesn't have signature nor compilation date

**Comment by radare** [comment]
Maybe we should use a three state value instead of bool to specify if the value is available or not

**Comment by radare** [comment]
relro in fact is not a boolean. read this http://tk-blog.blogspot.com.es/2009/02/relro-not-so-well-known-memory.html

we should tell if its partial or full. also , i think we can use this relro concept for other targets so its ok to merge for me once this issue is fixed.


**Comment by radare** [comment]
ping

**Comment by alvarofe** [comment]
relro must be three state (no relro, partial relro and full relro) that is the info is stored now in sdb and extracted with `ik`

File: libr/core/cbin.c [hunk 1]
```diff
@@ -673,58 +673,69 @@ static int bin_info(RCore *r, int mode) {
...
 		if (info->rclass && !strcmp (info->rclass, "pe")) {
-			pair_bool ("overlay", info->pe_overlay, mode, false);
 			//this should be moved if added to mach0 (or others)
 			pair_bool ("signed", info->signature, mode, false);
 		}
+		pair_bool ("static", r_bin_is_static (r->bin), mode, false);
+		pair_bool ("stripped", R_BIN_DBG_STRIPPED & info->dbg_info, mode, false);
+		pair_str ("subsys", info->subsystem, mode, false);
+		pair_bool ("va", info->has_va, mode, false);
+
```
Comment (radare) [inline libr/core/cbin.c]: so many empty lines

File: libr/core/cmd_info.c [hunk 2]
```diff
@@ -200,9 +200,10 @@ static void cmd_info_bin(RCore *core, int va, int mode) {
 		if (obj && bin_is_executable (obj)) {
 			if ((mode & R_CORE_BIN_JSON)) {
 				r_cons_printf (",\"bin\":");
+			} else {
+				r_cons_printf("\n");
```
Comment (radare) [inline libr/core/cmd_info.c]: missing space before (

**Comment by radare** [comment]
ping

**Comment by oddcoder** [comment]
https://github.com/radare/radare2-regressions/pull/781

**Comment by radare** [comment]
pls fix conflict

**Comment by radare** [comment]
and update tests

**Comment by radare** [comment]
tired of waiting

**Pull request merged**